exporter/context.go

```diff
@@ -509,13 +513,22 @@ func (ic *importContext) reference(i importable, path []string, value string) hc
 		if d.Match != "" {
 			attr = d.Match
 		}
-		traversal := ic.Find(&resource{
+		attrValue, traversal := ic.Find(&resource{
 			Resource:  d.Resource,
 			Attribute: attr,
 			Value:     value,
-		}, attr)
-		//at least one invocation of ic.Find will assign Nil to traversal if resource with value is not found
+		}, attr, d.MatchType)
+		// at least one invocation of ic.Find will assign Nil to traversal if resource with value is not found
 		if traversal != nil {
+			if d.MatchType == "prefix" { // we're replacing the found prefix with the HCL reference using the "${reference}rest" syntax
```

Comment:
is it more complex if `ic.Find` returns prefixed traversal?

Reply:
you mean to generate tokens directly in the `ic.Find` ? yes, maybe it's good idea to isolate it there